x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/InferencePluginTests.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference;
+
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.inference.services.elastic.ElasticInferenceServiceSettings;
+import org.junit.After;
+import org.junit.Before;
+
+import static org.hamcrest.Matchers.is;
+
+public class InferencePluginTests extends ESTestCase {
+    private InferencePlugin inferencePlugin;
+
+    private Boolean elasticInferenceServiceEnabled = true;
+
+    private void setElasticInferenceServiceEnabled(Boolean elasticInferenceServiceEnabled) {
+        this.elasticInferenceServiceEnabled = elasticInferenceServiceEnabled;
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+
+        Settings settings = Settings.builder().build();
+        inferencePlugin = new InferencePlugin(settings) {
+            @Override
+            protected Boolean isElasticInferenceServiceEnabled() {
+                return elasticInferenceServiceEnabled;
+            }
+        };
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+        // set default value for elasticInferenceServiceEnabled
+        setElasticInferenceServiceEnabled(true);
```

Comment:
Should we remove that and always call `setElasticInferenceServiceEnabled` in each test explicitly? Feels easier to read vs. having `tearDown` set state for a test